modules/core/test/src-js/smithy4s/JsConvertersSpec.scala

```diff
@@ -0,0 +1,54 @@
+package smithy4s
+
+import weaver._
+import scala.scalajs.js
+
+object JsConvertersSpec extends FunSuite {
+
+  test("object - strings") {
+    val input = js.Dictionary("one" -> "1")
+    val result = JsConverters.convertJsToDocument(input)
+    val expected =
+      Right(Document.DObject(Map("one" -> Document.fromString("1"))))
+    expect(result == expected)
+  }
+
+  test("array - booleans") {
+    val input = js.Array(true, false)
+    val result = JsConverters.convertJsToDocument(input)
+    val expected =
+      Right(Document.DArray(Vector(true, false).map(Document.fromBoolean)))
+    expect(result == expected)
+  }
+
+  test("integer") {
+    val input = 1
+    val result = JsConverters.convertJsToDocument(input)
+    val expected = Right(Document.fromInt(1))
+    expect(result == expected)
+  }
+
+  test("null") {
```

Comment:
Another one for undefined would be great here :)

Reply:
Good call, just added 🙂